drivers/libevbackend/src/libevbackend.cpp

```
@@ -390,18 +408,29 @@ void EventDevice::emitEvent(int type, int code, int value) {
 	};
 
 	// Filter out events that do not update the device state.
-	if(type == EV_KEY && getBit(_currentKeys.data(), _currentKeys.size(), code) == value)
-		return;
-	if(type == EV_REL && !value)
-		return;
-	if(type == EV_ABS && value == _absoluteSlots[code].value)
-		return;
+	switch(type) {
+	case EV_KEY:
+		if (getBit(_currentKeys.data(), _currentKeys.size(), code) == value)
+			return;
+		break;
+	case EV_REL:
+		if (!value)
+			return;
+		break;
+	case EV_ABS:
+		if (value == _absoluteSlots[code].value)
+			return;
+		break;
+	default:;
```

What is this `;` doing here?

---

Static code analyzers are complaining if no default clause is defined and it helps on code robustness. 
Specifying default:; you simply say other cases will not be taken into account.

---

The default clause I get and agree on, I'm just not familiar with adding a `;` directly behind it, because I always see a separate `break;` or an `assert` or actual code in the default statement, which is why I asked.

---

I'd say for default cases that should not be reached we should do something like `assert(!"Unexpected value");` (or maybe throw an exception).

---

Agreed, and even the coding style strongly suggests it.

---

This is a lesson learned after fixing QA C++ warnings : )